apps/src/code-studio/components/progress/TeacherStageInfo.jsx

```diff
@@ -79,13 +98,28 @@ const TeacherStageInfo = React.createClass({
             </span>
           }
           {lockable && <StageLock stage={stage}/>}
+          {experiments.isEnabled('hiddenStages') && <div style={styles.toggle}>
+            <HiddenStageToggle
+              hidden={this.props.isHidden}
+              onChange={this.onClickHiddenToggle}
+            />
+          </div>
+          }
         </div>
       </div>
     );
   }
 });
 
-export default connect(state => ({
-  hasNoSections: state.stageLock.sectionsLoaded &&
-    Object.keys(state.stageLock.sections).length === 0
+export default connect((state, ownProps) => {
+  const isHidden = state.hiddenStage[ownProps.stage.id];
+  return {
+    isHidden,
+    hasNoSections: state.stageLock.sectionsLoaded &&
+      Object.keys(state.stageLock.sections).length === 0
+    };
+}, dispatch => ({
```

Comment:
Not sure that I love that I'm increasingly using ownProps in connect. The other option here would be to provide the entire `state.hiddenStage` object as a prop here, and then do the lookup of this particular stageId in the render.